CitesphereConnector.py

```diff
@@ -0,0 +1,106 @@
+
+import urllib2
+import json
+
+# CITESPHERE_API = "https://citesphere.instance.org/api/v1"
+CITESPHERE_API = "https://diging-dev.asu.edu/citesphere-review/api"
+
+
+class CitesphereConnector:
+    def __init__(self, **kwargs):
+        self.client_id = kwargs.get('client_id')
+        self.client_secret = kwargs.get('client_secret')
+        self.grant_type = kwargs.get('grant_type')
+        self.redirect_uri = kwargs.get('redirect_uri')
+        self.authorization_code = kwargs.get('authorization_code')
+        self.api_key = None
+        self.access_token = kwargs.get('access_token')
```

Comment:
Instead of passing a bunch of method parameters, I would create classes for the different types of authentication. You pass either an object of an `OAuth2Authorization` or `BearerAccessTokenAuthorization`, etc. The headers should be created from these classes as well, since they might not always be using the `Bearer` prefix. This way we would only need to add new Authentication classes rather than changes the code of these methods whenever we change/add to the authorization process. Does that make sense?